src/user.rs

```diff
@@ -187,3 +195,33 @@ pub async fn user_main(
         );
     }
 }
+
+// Invoke the task function, collecting task statistics.
+async fn invoke_task_function(
+    function: &GooseTaskFunction,
+    thread_user: &GooseUser,
+    task_index: usize,
+    thread_task_name: &str,
+) {
+    let started = time::Instant::now();
+    let mut raw_task = GooseRawTask::new(
+        thread_user.started.elapsed().as_millis(),
+        thread_user.task_sets_index,
+        task_index,
+        thread_task_name.to_string(),
+        thread_user.weighted_users_index,
+    );
+    let success = function(&thread_user).await.is_ok();
+    raw_task.set_time(started.elapsed().as_millis(), success);
+
+    // Exit if all statistics or task statistics are disabled.
+    if thread_user.config.no_stats || thread_user.config.no_task_stats {
+        return;
+    }
+
+    // Otherwise send statistics to parent (which doesn't exist during testing).
```

Comment:
Should there be a mock parent for when those options are enabled?

Reply:
I actually think the comment is wrong; I'll test and fix.